src/disk.c

```diff
@@ -867,9 +867,13 @@ static int disk_read (void)
 		output_name = disk_name;
 
 #if HAVE_LIBUDEV
-		char *alt_name = disk_udev_attr_name (handle_udev, disk_name, conf_udev_name_attr);
-		if (alt_name != NULL)
-			output_name = alt_name;
+		char *alt_name = NULL;
+		if (conf_udev_name_attr != NULL)
+		{
+			alt_name = disk_udev_attr_name (handle_udev, disk_name, conf_udev_name_attr);
+			if (alt_name != NULL)
+				output_name = alt_name;
```

Comment:
This leaks alt_name if ignorelist_match is true and we continue the loop.
disk_udev_attr_name allocates, so the result should be freed.